torch/_dynamo/config.py

```diff
@@ -384,7 +384,7 @@ def _get_optimize_ddp_mode():
 enable_cpp_guard_manager = True
 
 # Use C++ guard manger for symbolic shapes
-enable_cpp_symbolic_shape_guards = False
+enable_cpp_symbolic_shape_guards = True
```

Comment:
Instead of `True`, we should perhaps have `not is_fbcode()`

cc @ezyang because I remember you telling me that internally we can't compile Cpp on the fly (though I am confused because we should have seen this problem with Inductor before_)

Reply:
we can compile C++ code on the fly but it has to go through RE. Anyway, OSS rollout first is a good idea anyway. @anijain2305 you will probably have to captain the FB rollout.